### PR TITLE
Update Eyes gem

### PR DESCRIPTION
**Pull request by breville**
This updates the Applitools `eyes_selenium` gem from 3.14.10 to 3.17.20.

It also makes some changes to Eyes tests that helped get them running against my localhost.

One notable regression with this update was that using either stitching method on a Droplet-based level, in RTL, and after toggling to code mode, it took a very wide screenshot with many screens' worth of blank white to the right of the real screen.  For now the workaround is to not use a stitching mode, so I've added the ability to specify a stitch mode of `"none"`.

We were previously seeing some flakiness in App Lab captures in which we'd sometimes have a small amount of additional white space below the page's UI.  This additional white space was showing consistently in captures done with this update, but using stitch mode `"none"` prevented it from happening, so I've added it to all the captures that exhibited the issue.

As can be seen in the [original commit](https://github.com/code-dot-org/code-dot-org/pull/36563/commits/674a88d6f3f6d0c9c0165ef48d934471d3e3ae73), updating the `eyes_selenium gem` to 3.17.20 also updated `selenium-webdriver` to 3.142.7, which led us to encountering [this issue](https://github.com/SeleniumHQ/selenium/issues/7158), and interfering with significant work done by @wjordan  in https://github.com/code-dot-org/code-dot-org/pull/28254.  This PR now does a [conservative update](https://bundler.io/man/bundle-install.1.html#CONSERVATIVE-UPDATING) to only update `eyes_selenium` without updating `selenium-webdriver`.

At some point we should consider updating our Eyes API usage to their newer [Fluent API.](https://applitools.com/docs/topics/sdk/the-eyes-sdk-check-fluent-api.html).

### CODE REVIEW

**Review by wjordan — APPROVED** [review]
I can't comment in detail on the UI test changes part (tests are passing which is good), but the Gem-upgrade part looks good to me! 👍

File: dashboard/test/ui/features/initial_page_views2.feature [hunk 1]
```diff
@@ -19,7 +19,7 @@ Feature: Looking at a few things with Applitools Eyes - Part 2
     And I sign out
     Examples:
       | url                                                      | test_name                         | stitch_mode | user_type |
-      | http://studio.code.org/projects/applab/new               | new applab project                | scroll      | student   |
+      | http://studio.code.org/projects/applab/new               | new applab project                | none        | student   |
```
Comment (maddiedierker) [inline dashboard/test/ui/features/initial_page_views2.feature:22]: i always forget you can do this (define a table of examples to run against)!

File: dashboard/test/ui/features/initial_page_views3.feature [hunk 2]
```diff
@@ -24,6 +24,7 @@ Feature: Looking at a few things with Applitools Eyes - Part 3
     Given I am on "http://studio.code.org/"
     When I open my eyes to test "<test_name>"
     And I am on "<url>"
+    And I dismiss the language selector
```
Comment (maddiedierker) [inline dashboard/test/ui/features/initial_page_views3.feature:27]: do you know why this step wasn't necessary before we upgraded the eyes gem?

**Review by maddiedierker — APPROVED** [review]
👀 🎉

File: dashboard/test/ui/features/initial_page_views3.feature (hunk 2 above)
Comment (breville) [reply dashboard/test/ui/features/initial_page_views3.feature:27]: I suspect it's not related to the gem upgrade, but rather that I was running these against my localhost, which is where the language selector was showing.  I didn't figure out the root cause, but suspect that these just don't get run against localhost very often.

**Pull request merged**